src/compiler/types.ts

```diff
@@ -562,6 +562,10 @@ namespace ts {
     export interface ShorthandPropertyAssignment extends ObjectLiteralElement {
         name: Identifier;
         questionToken?: Node;
+        // used when ObjectLiteralExpression is used in ObjectAssignmentPattern
+        // it is grammar error to appear in actual object initializer
+        equalsToken?: Node;

```

Comment:
Okay; in that case, can you amend the comment so it specifically says "The below are used". and just make it "It is a grammar error".

Other than that LGTM.